dumpgenerator.py

```diff
@@ -2188,7 +2188,7 @@ def resumePreviousDump(config={}, other={}):
 
```

Comment:
Can't we decode already in line 2183?

Reply:
As in this?

```python
            for l in f:
                l = l.decode('utf-8')
                if l == '</mediawiki>':
                # ...
```

I think so, though it probably would require other changes (`u'</mediawiki>'`, and maybe the regex would need to be changed too - I'm not sure how that works in python 2). It might also be possible to change `reverse_readline` to decode each line. I'd need to test; the current version of this PR is something I quickly hacked together to resume, but now that I'm not in the middle of a huge download I can try more things.

Reply:
Yes, something like that. Ok, please update this PR if you manage, otherwise let's merge it like this and maybe leave a TODO comment to remember to check.

Reply:
I've changed `reverse_readline`, and also fixed its truncation behavior. I tested using `python2 -u dumpgenerator.py --xml --api https://fr.wikiversity.org/w/api.php --force --namespace 11` (as there are only about 200 pages there, and the namespace is `Discussion modèle` which is easy enough to read but still contains unicode) and interrupting with <kbd>^C</kbd>.

Reply:
Thanks!